api/views.py

```diff
@@ -186,6 +186,15 @@ def partial_update(self, request, pk=None, **kwargs):
                     puzzle.status = data["status"]
                     puzzle.save()
 
+                if puzzle.chat_room and "name" in data:
+                    puzzle.chat_room.name = data["name"]
+                    puzzle.chat_room.save()
+                    transaction.on_commit(
+                        lambda: chat.tasks.handle_channel_rename.delay(
+                            puzzle.id, data["name"]
+                        )
```

Comment:
Not 100% sure about what the chat model expects but should we rename puzzle.chat_room after we actually change the discord channel name successfully in the task?

Reply:
As discussed offline -- `chat_room.name` is actually duplicative with puzzle.name. We should probably figure out if we actually need this (and delete it if not).